ethcore/src/blockchain/blockchain.rs

```diff
@@ -753,8 +770,16 @@ impl BlockChain {
 
 				// TODO: handle block_hashes properly.
 				block_hashes.clear();
+
+				blocks.shrink_to_fit();
```

Comment:
Removing something from a `HashMap` does not guarantee the memory will be deallocated. We use `heapsize` to measure cache size on line 782. The condition there would always be false after max size is reached for the first time.